funcframework/framework.go

```diff
@@ -97,61 +99,95 @@ func RegisterHTTPFunctionContext(ctx context.Context, path string, fn func(http.
 // context.Context and a struct type depending on the event, and return an error. If fn has the
 // wrong signature, RegisterEventFunction returns an error.
 func RegisterEventFunctionContext(ctx context.Context, path string, fn interface{}) error {
-	server, err := wrapEventFunction(path, fn)
+	handler, err := wrapEventFunction(fn)
 	if err == nil {
-		handler = server
+		server.Handle(path, handler)
+		handlerRegistered = true
 	}
 	return err
 }
 
 // RegisterCloudEventFunctionContext registers fn as an cloudevent function.
 func RegisterCloudEventFunctionContext(ctx context.Context, path string, fn func(context.Context, cloudevents.Event) error) error {
-	server, err := wrapCloudEventFunction(ctx, path, fn)
+	handler, err := wrapCloudEventFunction(ctx, fn)
 	if err == nil {
-		handler = server
+		server.Handle(path, handler)
+		handlerRegistered = true
 	}
 	return err
 }
 
 // Start serves an HTTP server with registered function(s).
 func Start(port string) error {
-	// If FUNCTION_TARGET, try to start with that registered function
-	// If not set, assume non-declarative functions.
-	target := os.Getenv("FUNCTION_TARGET")
-
-	// Check if we have a function resource set, and if so, log progress.
-	if os.Getenv("K_SERVICE") == "" {
-		fmt.Printf("Serving function: %s\n", target)
+	if err := initServer(); err != nil {
+		return err
 	}
+	return http.ListenAndServe(":"+port, server)
+}
 
-	// Check if there's a registered function, and use if possible
-	if fn, ok := registry.Default().GetRegisteredFunction(target); ok {
-		ctx := context.Background()
-		if fn.HTTPFn != nil {
-			server, err := wrapHTTPFunction("/", fn.HTTPFn)
-			if err != nil {
-				return fmt.Errorf("unexpected error in registerHTTPFunction: %v", err)
-			}
-			handler = server
-		} else if fn.CloudEventFn != nil {
-			server, err := wrapCloudEventFunction(ctx, "/", fn.CloudEventFn)
+func initServer() error {
+	target := os.Getenv("FUNCTION_TARGET")
+	switch {
+	// If FUNCTION_TARGET is set, only serve this target function.
+	case len(target) > 0:
+		// Start a new server because the same route "/" can only be
```

Comment:
Can we avoid this problem by deferring `server.Handle` call until this function? We can keep a file-global `handlerPath` or something instead of `handlerRegistered`?